### PR TITLE
Fix window can't be hidden when maximized

### PR DESCRIPTION
**Pull request by wusyong**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

**Comment by amrbashir** [comment]
Handling `WindowFlags::MAXIMIZED` before `WindowFlags::VISIBLE`  will bring back an old bug where maximizing a hidden window will also show it.  Any call to `ShowWindow` should be after handling `WindowFlags::VISIBLE`

**Comment by wusyong** [comment]
How about I check it  on MAXIMIZED flag diff instead?

**Comment by amrbashir** [comment]
It works but it has a bug: 
1. maximize
2. hide
3. show
4. unmaximize -> the window doesn't return to its former size

File: src/platform_impl/windows/window_state.rs [hunk 1]
```diff
@@ -239,7 +239,19 @@ impl WindowFlags {
     self = self.mask();
     new = new.mask();
 
-    let diff = self ^ new;
+    let mut diff = self ^ new;
+
+    // when hiding a maximized window, `self` contains `WindowFlags::MAXIMIZED`
```
Comment (lucasfernog) [inline src/platform_impl/windows/window_state.rs:249]:
```suggestion
    // when hiding a maximized window, `diff` contains `WindowFlags::MAXIMIZED`
```

**Comment by wusyong** [comment]
The update is working for me. @lucasfernog could you also check again?

**Comment by lucasfernog** [comment]
I can confirm this is working now.

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**